akka-typed/src/main/scala/akka/typed/ActorContext.scala

```diff
@@ -9,136 +9,51 @@ import akka.util.Helpers
 import akka.{ actor ⇒ untyped }
 import scala.concurrent.duration.FiniteDuration
 import scala.concurrent.ExecutionContextExecutor
+import java.util.Optional
+import java.util.ArrayList
 
 /**
- * An Actor is given by the combination of a [[Behavior]] and a context in
- * which this behavior is executed. As per the Actor Model an Actor can perform
- * the following actions when processing a message:
- *
- *  - send a finite number of messages to other Actors it knows
- *  - create a finite number of Actors
- *  - designate the behavior for the next message
- *
- * In Akka the first capability is accessed by using the `!` or `tell` method
- * on an [[ActorRef]], the second is provided by [[ActorContext#spawn]]
- * and the third is implicit in the signature of [[Behavior]] in that the next
- * behavior is always returned from the message processing logic.
- *
- * An `ActorContext` in addition provides access to the Actor’s own identity (“`self`”),
- * the [[ActorSystem]] it is part of, methods for querying the list of child Actors it
- * created, access to [[Terminated DeathWatch]] and timed message scheduling.
+ * This trait is not meant to be extended by user code. If you do so, you may
+ * lose binary compatibility.
```

Comment:
We have some annotations that you also can start using.
`@DoNotInherit`
`@InternalApi`
`@ApiMayChange` (the whole module has this status so adding it to a few top classes and scala package is enough)